docs/source/visualize-with-clusters.md

```diff
@@ -0,0 +1,78 @@
+# Visualize a Model with Highlighted QDQ Clusters and Partitions
+
+The [visualization_utils.py](/executorch/devtools/visualization/visualization_utils.py) contains the function
+`visualize_with_clusters()` which takes an `ExportedProgram` and visualizes it using the `ModelExplorer` utility.  
+It groups QDQ clusters and individual partitions together to improve readability.
+
+---
+
+## Installation
+
+To install the `ModelExplorer` and its dependencies, run:
+
+```bash
+$ ./devtools/install_requirements.sh
+```
+
+---
+
+## Usage
+
+There are two main use cases for the visualization:
+
+### 1. Launching the `ModelExplorer` and Visualizing the Model Immediately
+
+Call:
+
+```python
+visualize_with_clusters(exported_program)
+```
+
+This starts a `ModelExplorer` server and opens a browser tab with the visualization.
+
+By default, each call starts a new server instance and opens a new browser tab.  
+To reuse an existing server, set the `reuse_server` parameter to `True`.
+
+Starting the server is **blocking**, so the rest of your script will not run.
+
+### 2. Storing a Serialized Graph and Visualizing Later (Non-blocking)
+
+To save the visualization to a JSON file, call:
+
+```python
+visualize_with_clusters(exported_program, "my_model.json")
+```
+
+This just saves the visualization in the file, and it does **not** start the `ModelExplorer` server. You can then open
+the file in the `ModelExplorer` GUI at any point. To launch the server, run:
+
+```bash
+$ model-explorer
+```
+
+A browser tab should open automatically.  
+Click **Select from your computer**, choose the JSON file, and then click **View selected models** to display the graph.
+
+---
+
+## Styling the Graph
+
+`visualize_with_clusters()` supports custom grouping of nodes into QDQ clusters and partitions.
+
+You can pass the following optional parameters:
+
+- `get_node_partition_name`
+- `get_node_qdq_cluster_name`
+
+These are functions that take a node and return a string identifying the partition or cluster it belongs to.  
+Nodes with the same partition/cluster string will be grouped together and labeled accordingly in the visualization.
+
+### Custom Styles in `ModelExplorer`
+
+To customize the appearance of nodes:
```

Comment:
There is already a color style for the QDQ cluster and partitions highlighting provided in the `devtools/visualization/model_explorer_styles/cluster_highlight_style.json`. To load it follow these steps: